JavadocTagContinuationIndentation check doesn't support arbitrary indentation on &lt;pre&gt; tags

I have read check documentation: https://checkstyle.org/config_javadoc.html#JavadocTagContinuationIndentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ javac Test.java

$ cat test-checkstyle.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
  <module name="TreeWalker">
    <module name="JavadocTagContinuationIndentation"/>
  </module>
</module>

$ cat Test.java 
/**
 * Test class.
 *
 * @apiNote
 *          This is the predefined indentation applied by Eclipse formatter to api/impl taglets.
 *          <pre>
 * Since &lt;pre> blocks retain leading whitespace, in order to avoid extra-indentation from being
 * added by the javadoc renderer, its content herein has to stick to the left margin of the comment
 * area. Unfortunately, checkstyle seems not to properly evaluate the context (&lt;pre> block) of
 * this content, and reports an error:
 *   "JavadocTagContinuationIndentation: Line continuation have incorrect indentation level,
 *   expected level should be 4"</pre>
 */
public class Test {
  public static void main(String[] args) {
  }
}

$ java -jar checkstyle-10.10.0-all.jar -c test-checkstyle.xml Test.java
Starting audit...
[ERROR] Test.java:7: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
[ERROR] Test.java:8: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
. . .
[ERROR] Test.java:12: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
Checkstyle ends with 6 errors.
```
---
CURRENT BEHAVIOR
Since &lt;pre&gt; blocks retain leading whitespace, in order to avoid extra-indentation from being added by the javadoc renderer, I have to stick their contents to the left margin of the comment area (like in `Test.java` here above) — here it is the intended result:
![Test-javadoc](https://user-images.githubusercontent.com/10812966/236636387-e7dab00a-32d7-46ad-bce2-93f001b1dad0.jpg)

Unfortunately, _checkstyle seems not to properly evaluate the context (i.e., it is unable to detect such contents are in a pre-formatted block which should be excluded from indentation enforcement)_, and raises errors of non-continuous indentation.

BTW, if, conversely, I conform the indentation of the pre-formatted block to the checkstyle rule this way...
```bash
/**
 * Test class.
 *
 * @apiNote
 *          This is the predefined indentation applied by Eclipse formatter to api/impl taglets.
 *          <pre>
 *          Since &lt;pre> blocks retain leading whitespace, in order to avoid extra-indentation from being
 *          added by the javadoc renderer, its content herein has to stick to the left margin of the comment
 *          area. Unfortunately, checkstyle seems not to properly evaluate the context (&lt;pre> block) of
 *          this content, and reports an error:
 *            "JavadocTagContinuationIndentation: Line continuation have incorrect indentation level,
 *            expected level should be 4"</pre>
 */
public class Test_extraindent {
  public static void main(String[] args) {
  }
}
```
...the resulting HTML rendering is —unsurprisingly— unacceptably over-indented (the same happens when rendered directly from source code in the Javadoc view of Eclipse IDE):
![Test_extraindent-javadoc](https://user-images.githubusercontent.com/10812966/236636453-9ee2b4b5-c64b-44fb-a6e8-c75fbf3745a2.jpg)

---
EXPECTED BEHAVIOR
Because of their inherent semantics, **pre-formatted blocks should be allowed arbitrary indentation**: once checkstyle detects an opening &lt;pre&gt; tag, it should automatically disable the `JavadocTagContinuationIndentation` check until it reaches its corresponding closing &lt;/pre&gt; tag.

how to make java doc comment to pass javadoc tool ?
```
$ cat Test.java 
/**
 * <p>Java class for errorResponse complex type.
 *
 * <p>The following schema fragment specifies the expected content contained within this class.
 *
 * @apiNote xml of that match xsd schema
 * <pre>
 * &lt;complexType name="errorResponse"&gt;
 *   &lt;complexContent&gt;
 *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType"&gt;
 *       &lt;sequence&gt;
 *         &lt;element name="errors" minOccurs="0"&gt;
 *           &lt;complexType&gt;
 *             &lt;complexContent&gt;
 *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType"&gt;
 *                 &lt;sequence&gt;
 *                   &lt;element name="error" type="{http://www.w3.org/2001/XMLSchema}string" maxOccurs="unbounded" minOccurs="0"/&gt;
 *                 &lt;/sequence&gt;
 *               &lt;/restriction&gt;
 *             &lt;/complexContent&gt;
 *           &lt;/complexType&gt;
 *         &lt;/element&gt;
 *       &lt;/sequence&gt;
 *     &lt;/restriction&gt;
 *   &lt;/complexContent&gt;
 * &lt;/complexType&gt;
 * </pre>
 *
 *
 */
public class Test {

    protected String errors;
}rivanov@p5510:/var/tmp$ javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Standard Doclet version 11.0.14.1
Building tree for all the packages and classes...
Test.java:6: error: unknown tag: apiNote
 * @apiNote xml of that match xsd schema
   ^
Generating ./Test.html...
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./constant-values.html...
Building index for all the packages and classes...
Generating ./overview-tree.html...
Generating ./index-all.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./deprecated-list.html...
Building index for all classes...
Generating ./allclasses.html...
Generating ./allclasses.html...
Generating ./index.html...
Generating ./help-doc.html...
1 error

``` 

@stechio , do you know ?

--- a/config/jsoref-spellchecker/whitelist.words
+++ b/config/jsoref-spellchecker/whitelist.words
@@ -913,7 +913,7 @@ mytag
 NAMEFILE
 nameofaffectedcheck
 namespace
-Nanos
+nanos
 nbactions
 nbia
 nbsp

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java
@@ -36,6 +36,12 @@ import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
  * the tag. Default indentation required is at least 4, but this can be changed with the help of
  * properties below.
  * </div>
+ * <ul>
+ * <li>
+ * Notes:
+ * This check does not validate the indentation of lines inside {@code pre} tags.
+ * </li>
+ * </ul>
  *
  * @since 6.0
  */
@@ -81,13 +87,43 @@ public class JavadocTagContinuationIndentationCheck extends AbstractJavadocCheck
     public void visitJavadocToken(DetailNode ast) {
         if (isBlockDescription(ast) && !isInlineDescription(ast)) {
             final List<DetailNode> textNodes = getAllNewlineNodes(ast);
+            boolean isTextPreTagChildren = false;
             for (DetailNode newlineNode : textNodes) {
                 final DetailNode textNode = JavadocUtil.getNextSibling(newlineNode);
-                if (textNode.getType() != JavadocTokenTypes.NEWLINE && isViolation(textNode)) {
+                if (newlineNode.getType() == JavadocTokenTypes.HTML_ELEMENT_START
+                        || isHtmlTagChildOfPreTag(ast)) {
+                    isTextPreTagChildren = true;
+                }
+                else if (newlineNode.getType() == JavadocTokenTypes.HTML_ELEMENT_END) {
+                    isTextPreTagChildren = false;
+                }
+                else if (!isTextPreTagChildren && textNode.getType() != JavadocTokenTypes.NEWLINE
+                    && isViolation(textNode)) {
                     log(textNode.getLineNumber(), MSG_KEY, offset);
                 }
             }
         }
+    }
+
+    /**
+     * Checks if the given HTML_TAG is contained inside {@code <pre>} tag.
+     * For cases when another HTML tag is placed next to or before {@code <pre>} tag.
+     * For example:
+     * <pre>
+     * {@code
+     * <pre><someOtherTag>
+     *     some thing
+     * </someOtherTag></pre>
+     * }
+     * </pre>
+     *
+     * @param htmlTag HTML_TAG
+     * @return {@code true} if {@code pre} tag is parent of the given tag, else {@code false}.
+     */
+    private static boolean isHtmlTagChildOfPreTag(DetailNode htmlTag) {
+        DetailNode node = htmlTag.getParent().getParent();
+        node = JavadocUtil.getFirstChild(node);
+        return containsPreTag(node);
     }
 
     /**
@@ -106,8 +142,9 @@ public class JavadocTagContinuationIndentationCheck extends AbstractJavadocCheck
         if (text.length() <= offset) {
             if (CommonUtil.isBlank(text)) {
                 final DetailNode nextNode = JavadocUtil.getNextSibling(textNode);
-                if (nextNode != null && nextNode.getType() != JavadocTokenTypes.NEWLINE) {
-                    // text is blank but line hasn't ended yet
+                // text is blank but line hasn't ended yet
+                if (nextNode != null && nextNode.getType() != JavadocTokenTypes.NEWLINE
+                        && !containsPreTag(nextNode)) {
                     result = true;
                 }
             }
@@ -139,6 +176,9 @@ public class JavadocTagContinuationIndentationCheck extends AbstractJavadocCheck
             }
             else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT_START
                 || node.getType() == JavadocTokenTypes.ATTRIBUTE) {
+                if (containsPreTag(node)) {
+                    textNodes.add(node);
+                }
                 textNodes.addAll(getAllNewlineNodes(node));
             }
             if (node.getType() == JavadocTokenTypes.LEADING_ASTERISK) {
@@ -146,7 +186,35 @@ public class JavadocTagContinuationIndentationCheck extends AbstractJavadocCheck
             }
             node = JavadocUtil.getNextSibling(node);
         }
+
+        // Last node does not get checked in the loop
+        if (containsPreTag(node)) {
+            textNodes.add(node);
+        }
         return textNodes;
+    }
+
+    /**
+     * Checks if the given HTML related node contains {@code <pre>} tag.
+     *
+     * @param ast the node to check
+     * @return {@code true} if the {@code <pre>} tag is inside the node, {@code false} otherwise
+     */
+    private static boolean containsPreTag(DetailNode ast) {
+        DetailNode node = ast;
+        if (node.getType() == JavadocTokenTypes.HTML_ELEMENT_START
+                || node.getType() == JavadocTokenTypes.HTML_ELEMENT_END) {
+            node = JavadocUtil.findFirstToken(node, JavadocTokenTypes.HTML_TAG_NAME);
+        }
+        if (node.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+            final DetailNode htmlTag = JavadocUtil.getFirstChild(node);
+            if (htmlTag.getType() == JavadocTokenTypes.HTML_TAG) {
+                final DetailNode htmlElementStart = JavadocUtil.getFirstChild(htmlTag);
+                node = JavadocUtil.findFirstToken(htmlElementStart,
+                    JavadocTokenTypes.HTML_TAG_NAME);
+            }
+        }
+        return "pre".equalsIgnoreCase(node.getText());
     }
 
     /**

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/javadoc/JavadocTagContinuationIndentationCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/javadoc/JavadocTagContinuationIndentationCheck.xml
@@ -10,7 +10,13 @@
  throws a violation. A continuation line is when the description starts/spans past the line with
  the tag. Default indentation required is at least 4, but this can be changed with the help of
  properties below.
- &lt;/div&gt;</description>
+ &lt;/div&gt;
+
+ &lt;ul&gt;
+ &lt;li&gt;
+ This check does not validate the indentation of lines inside &lt;code&gt;pre&lt;/code&gt; tags.
+ &lt;/li&gt;
+ &lt;/ul&gt;</description>
          <properties>
             <property default-value="4" name="offset" type="int">
                <description>Specify how many spaces to use for new indentation level.</description>

--- a/src/site/xdoc/checks/javadoc/javadoctagcontinuationindentation.xml
+++ b/src/site/xdoc/checks/javadoc/javadoctagcontinuationindentation.xml
@@ -18,6 +18,14 @@
         </div>
       </subsection>
 
+      <subsection name="Notes" id="JavadocTagContinuationIndentation_Notes">
+        <ul>
+        <li>
+          This check does not validate the indentation of lines inside <code>pre</code> tags.
+        </li>
+        </ul>
+      </subsection>
+
       <subsection name="Properties" id="JavadocTagContinuationIndentation_Properties">
         <div class="wrapper">
           <table>
@@ -161,6 +169,49 @@ class Example3 {
   public void testMethod2(String input) {
     // violation 3 lines above 'Line continuation have incorrect indentation level'
   }
+}
+</code></pre></div><hr class="example-separator"/>
+        <p id="Example4-code">
+         Check does not validate the indentation of lines inside <code>pre</code> tags.
+         Example:
+        </p>
+        <div class="wrapper"><pre class="prettyprint"><code class="language-java">
+class Example4 {
+
+   /**
+   * Writes the object using a
+   * &lt;a href="{@docRoot}/serialized-form.html#java.time.Ser"&gt;dedicated form&lt;/a&gt;.
+   * @serialData
+   * &lt;pre&gt;
+   * out.writeByte(1);
+   * out.writeInt(nanos);
+   * &lt;/pre&gt;
+   */
+  public void testMethod1(String input) {}
+
+  /**
+   * Writes the object using a
+   * &lt;a href="{@docRoot}/serialized-form.html#java.time.Ser"&gt;dedicated form&lt;/a&gt;.
+   * @serialData
+   * &lt;code&gt; // violation
+   * out.writeByte(1); // violation
+   * out.writeInt(nanos); // violation
+   * &lt;/code&gt; // violation
+   */
+  public void testMethod2(String input) {}
+
+  /**
+   * Test class.
+   *
+   * @apiNote
+   *          This is the predefined indentation applied by Eclipse formatter.
+   *          &lt;pre&gt;
+   * this content, and not any error:
+   *   "JavadocTagContinuation do not validate lines contained in Pre tag,
+   *   No violation is expected here."&lt;/pre&gt;
+   */
+  public void testMethod3(String input) {}
+
 }
 </code></pre></div>
       </subsection>

--- a/src/site/xdoc/checks/javadoc/javadoctagcontinuationindentation.xml.template
+++ b/src/site/xdoc/checks/javadoc/javadoctagcontinuationindentation.xml.template
@@ -15,6 +15,13 @@
         </macro>
       </subsection>
 
+      <subsection name="Notes" id="JavadocTagContinuationIndentation_Notes">
+        <macro name="notes">
+            <param name="modulePath"
+                   value="src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java"/>
+        </macro>
+      </subsection>
+
       <subsection name="Properties" id="JavadocTagContinuationIndentation_Properties">
         <div class="wrapper">
           <macro name="properties">
@@ -71,6 +78,15 @@
         <macro name="example">
           <param name="path"
                  value="resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/Example3.java"/>
+          <param name="type" value="code"/>
+        </macro><hr class="example-separator"/>
+        <p id="Example4-code">
+         Check does not validate the indentation of lines inside <code>pre</code> tags.
+         Example:
+        </p>
+        <macro name="example">
+          <param name="path"
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/Example4.java"/>
           <param name="type" value="code"/>
         </macro>
       </subsection>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheckTest.java
@@ -110,10 +110,8 @@ public class JavadocTagContinuationIndentationCheckTest
             "32: " + getCheckMessage(MSG_KEY, 4),
             "42: " + getCheckMessage(MSG_KEY, 4),
             "62: " + getCheckMessage(MSG_KEY, 4),
-            "64: " + getCheckMessage(MSG_KEY, 4),
             "74: " + getCheckMessage(MSG_KEY, 4),
             "75: " + getCheckMessage(MSG_KEY, 4),
-            "76: " + getCheckMessage(MSG_KEY, 4),
             "86: " + getCheckMessage(MSG_KEY, 4),
             "87: " + getCheckMessage(MSG_KEY, 4),
             "88: " + getCheckMessage(MSG_KEY, 4),
@@ -176,5 +174,53 @@ public class JavadocTagContinuationIndentationCheckTest
         verifyWithInlineConfigParser(
                 getPath("InputJavadocTagContinuationIndentationCheckHtml.java"),
                 expected);
+    }
+
+    @Test
+    public void testJavadocTagContinuationIndentationCheckPreTag() throws Exception {
+        final String[] expected = {
+            "91: " + getCheckMessage(MSG_KEY, 4),
+            "100: " + getCheckMessage(MSG_KEY, 4),
+            "101: " + getCheckMessage(MSG_KEY, 4),
+            "102: " + getCheckMessage(MSG_KEY, 4),
+        };
+        verifyWithInlineConfigParser(
+            getPath("InputJavadocTagContinuationIndentationPreTag.java"), expected);
+    }
+
+    @Test
+    public void testJavadocTagContinuationIndentationCheckPreTag2() throws Exception {
+        final String[] expected = {
+            "32: " + getCheckMessage(MSG_KEY, 4),
+            "33: " + getCheckMessage(MSG_KEY, 4),
+            "34: " + getCheckMessage(MSG_KEY, 4),
+            "35: " + getCheckMessage(MSG_KEY, 4),
+            "70: " + getCheckMessage(MSG_KEY, 4),
+            "78: " + getCheckMessage(MSG_KEY, 4),
+            "79: " + getCheckMessage(MSG_KEY, 4),
+            "80: " + getCheckMessage(MSG_KEY, 4),
+            "81: " + getCheckMessage(MSG_KEY, 4),
+            "82: " + getCheckMessage(MSG_KEY, 4),
+            "83: " + getCheckMessage(MSG_KEY, 4),
+            "84: " + getCheckMessage(MSG_KEY, 4),
+            "85: " + getCheckMessage(MSG_KEY, 4),
+            "90: " + getCheckMessage(MSG_KEY, 4),
+            "91: " + getCheckMessage(MSG_KEY, 4),
+            "92: " + getCheckMessage(MSG_KEY, 4),
+        };
+        verifyWithInlineConfigParser(
+            getPath("InputJavadocTagContinuationIndentationPreTag2.java"), expected);
+    }
+
+    @Test
+    public void testJavadocTagContinuationIndentationCheckPreTag3() throws Exception {
+        final String[] expected = {
+            "34: " + getCheckMessage(MSG_KEY, 4),
+            "35: " + getCheckMessage(MSG_KEY, 4),
+            "36: " + getCheckMessage(MSG_KEY, 4),
+            "37: " + getCheckMessage(MSG_KEY, 4),
+        };
+        verifyWithInlineConfigParser(
+            getPath("InputJavadocTagContinuationIndentationPreTag3.java"), expected);
     }
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java
@@ -61,7 +61,7 @@ public class InputJavadocTagContinuationIndentationBlockTag {
      * @param args
      * {@code this} line is not correctly indented // violation
      *     {@code this}
-     * <pre>this line is not correctly indented</pre> // violation
+     * <pre>pre tags are skipped</pre>
      */
     public void multipleLines1(String args) {
         // do stuff
@@ -73,7 +73,7 @@ public class InputJavadocTagContinuationIndentationBlockTag {
      * @return false always
      * {@code this} line is not correctly indented // violation
      * {@code this} line is not correctly indented // violation
-     * <pre>this line is not correctly indented</pre> // violation
+     * <pre>pre tags are skipped</pre>
      */
     public boolean isMultipleLines2() {
         return false;

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationPreTag.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationPreTag.java
@@ -1,0 +1,115 @@
+/*
+JavadocTagContinuationIndentation
+violateExecutionOnNonTightHtml = (default)false
+offset = (default)4
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadoctagcontinuationindentation;
+
+/**
+ * Test class.
+ *
+ * @apiNote
+ *          This is the predefined indentation applied by Eclipse formatter to api/impl taglets.
+ *          <pre>
+ * Since &lt;pre> blocks retain leading whitespace, in order to avoid extra-indentation from being
+ * added by the javadoc renderer, its content herein has to stick to the left margin of the comment
+ * area. Unfortunately, checkstyle seems not to properly evaluate the context (&lt;pre> block) of
+ * this content, and reports an error:
+ *   "JavadocTagContinuationIndentation: Line continuation have incorrect indentation level,
+ *   expected level should be 4"</pre>
+ */
+public class InputJavadocTagContinuationIndentationPreTag {
+
+    /**
+     * Test class.
+     *
+     * @apiNote
+     *          This is the predefined indentation applied by Eclipse formatter to api/impl taglets.
+     *          <pre>
+     *          Since &lt;pre> blocks retain leading whitespace, in order to avoid
+     *          the context (&lt;pre> block) of this content, and reports an error:
+     *            "JavadocTagContinuationIndentation: Line continuation have
+     *            incorrect indentation level,
+     *            expected level should be 4"</pre>
+     */
+    public void test1() {}
+
+    /**
+     * <p>Java class for errorResponse complex type.
+     *
+     * <p>The following schema fragment specifies the expected content contained within class.
+     *
+     * <pre>
+     * &lt;complexType name="errorResponse"&gt;
+     *   &lt;complexContent&gt;
+     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType"&gt;
+     *       &lt;sequence&gt;
+     *         &lt;element name="errors" minOccurs="0"&gt;
+     *           &lt;complexType&gt;
+     *             &lt;complexContent&gt;
+     *               &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType"&gt;
+     *                 &lt;sequence&gt;
+     *                   &lt;element name="error" type="{http://www.w3.org/2001/XMLSchema}str
+     *                 &lt;/sequence&gt;
+     *               &lt;/restriction&gt;
+     *             &lt;/complexContent&gt;
+     *           &lt;/complexType&gt;
+     *         &lt;/element&gt;
+     *       &lt;/sequence&gt;
+     *     &lt;/restriction&gt;
+     *   &lt;/complexContent&gt;
+     * &lt;/complexType&gt;
+     * </pre>
+     *
+     *
+     */
+    public void test2() {}
+
+    /**
+     * Test class.
+     *
+     * @apiNote
+     *          This is the predefined indentation applied by Eclipse formatter to api taglets.
+     *          <pre>
+     *   "JavadocTagContinuationIndentation: Line continuation have incorrect indentation level,
+     *   expected level should be 4"</pre> some text after the pre tag to check how the current
+     *     behaviour of the check.
+     */
+    public void test3() {}
+
+    /**
+     * Test class.
+     *
+     * @apiNote
+     *          This is the predefined indentation applied by Eclipse formatter to api taglets.
+     *          <pre>
+     *   "JavadocTagContinuationIndentation: Line continuation have incorrect indentation level,
+     *   expected level should be 4"</pre> some text after the pre tag to check how the current
+     *   behaviour of the check.
+     */
+    public void test4() {}
+    // violation 3 lines above 'Line continuation have .* expected level should be 4'
+
+    /**
+     * Writes the object using a
+     * <a href="{@docRoot}/serialized-form.html#ja>deserialized form</a>.
+     * @serialData
+     * Refer to the serialized form of // violation
+     * <a href="{@docRoot}/serialized-formRules">ZoneRules.writeReplace</a> // violation
+     * for the encoding of epoch seconds and offsets. // violation
+     * <pre style="font-size:1.0em">{@code
+     *
+     *   out.writeByte(2);                // identifies a ZoneOffsetTransition
+     *   out.writeEpochSec(toEpochSecond);
+     *   out.writeOffset(offsetAfter);
+     * }
+     * </pre>
+     * @return the replacing object, not null
+     */
+    private Object writeReplace() {
+        return new Object();
+    }
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationPreTag2.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationPreTag2.java
@@ -1,0 +1,118 @@
+/*
+JavadocTagContinuationIndentation
+violateExecutionOnNonTightHtml = (default)false
+offset = (default)4
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadoctagcontinuationindentation;
+
+import java.io.Serial;
+
+/**
+ * Test class.
+ *
+ * @apiNote
+ *          This is the predefined indentation applied by Eclipse formatter to api/impl taglets.
+ *          <pre>
+ * this content, and reports an error:
+ *   "JavadocTagContinuationIndentation: Line continuation have incorrect indentation level,
+ *   expected level should be 4"</pre>
+ */
+public class InputJavadocTagContinuationIndentationPreTag2 {
+
+     /**
+      * Writes implicitly tagged data using buffer-to-buffer copy.
+      *
+      * @param tag the DER value of the context-specific tag that replaces
+      *     original tag of the value in the output, such as in
+      * <pre>
+      *          <em> {@code <field> [N] IMPLICIT <type>}</em>
+      * </pre>
+      * For example, <em>FooLength [1] IMPLICIT INTEGER</em>, with value=4; // violation
+      * would be encoded as "81 01 04"  whereas in explicit // violation
+      * tagging it would be encoded as "A1 03 02 01 04". // violation
+      * explicit tagging the form is always constructed. // violation
+      * @param value original value being implicitly tagged
+      */
+     public Object writeImplicit(byte tag, Object value) {
+         return new Object();
+     }
+
+     /**
+      * Writes the object using a
+      * <a href="{@docRoot}/serialized-form.html#java.time.Ser">dedicated serialized form</a>.
+      * @serialData
+      * <pre>
+      *  out.writeByte(1);  // identifies a Duration
+      *  out.writeLong(seconds);
+      *  out.writeInt(name);
+      * </pre>
+      *
+      * @return the instance of {@code Ser}, not null
+      */
+     @Serial
+     private Object writeReplace() {
+         return new Object();
+     }
+
+     /**
+      * Queries this date-time.
+      * <p>
+      * This queries this date-time using the specified query strategy object.
+      * <p>
+      * Queries are a key tool for extracting information from date-times.
+      * <p>
+      * The most common query implementations are method references, such as
+      * Additional implementations are provided as static methods on {@link TemporalQuery}.
+      *
+      * @implSpec
+      * The default implementation must behave equivalent to this code: // violation
+      * <pre>
+      *  if (query == TemporalQueries.zoneId() ||
+      *        query == TemporalQueries.chronology() || query == TemporalQueries.precision()) {
+      *    return null;
+      *  }
+      *  return query.queryFrom(this);
+      * </pre>
+      * Future versions are permitted to add further queries to the if statement. // violation
+      * <p> // violation
+      * All classes implementing this interface and overriding this method must call // violation
+      * {@code TemporalAccessor.super.query(query)}. JDK classes may avoid calling // violation
+      * non-JDK classes may this optimization and must call {@code super}. // violation
+      * <p> // violation
+      * If the implementation for one of the queries listed in the // violation
+      * if statement of the default implementation, then it must do so. // violation
+      * <pre>
+      *    return MINUTES;
+      *  return TemporalAccessor.super.query(query);
+      * </pre>
+      * <p> // violation
+      * Implementations must ensure that no observable state is altered when this // violation
+      * read-only method is invoked. // violation
+      *
+      * @param query  the query to invoke, not null
+      */
+     Object query(Object query) {
+         return new Object();
+     }
+
+     /**
+      * Sends an attributeChangeNotification which contains the old value and new value for the
+      * attribute to the registered AttributeChangeNotification listeners on the ModelMBean.
+      *
+      * @param oldValue The original value for the Attribute
+      * @param newValue The current value for the Attribute
+      * <PRE>
+      * The constructed attributeChangeNotification will be:
+      * attributeType oldValue's class
+      *          attributeOldValue oldValue.getValue()
+      *   attributeNewValue newValue.getValue()
+      * </PRE>
+      *
+      * @exception MBeanException Wraps a distributed communication Exception.
+      */
+     Object query2(Object query) {
+         return new Object();
+     }
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationPreTag3.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationPreTag3.java
@@ -1,0 +1,43 @@
+/*
+JavadocTagContinuationIndentation
+violateExecutionOnNonTightHtml = (default)false
+offset = (default)4
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadoctagcontinuationindentation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+public class InputJavadocTagContinuationIndentationPreTag3 {
+
+    @Target(ElementType.TYPE_USE)
+    @interface Internal {}
+
+    /**
+     * @return sub role of this role, which match `item`.
+     *
+     * <pre><code>
+     * CtMethod method = ...
+     * CtRole role = CtRole.TYPE_MEMBER.getMatchingSubRoleFor(method);
+     * </code></pre>
+     */
+    @Internal
+    public Object testMethod() {
+        return new Object();
+    }
+
+    /**
+     * @return sub role of this role, which match `item`.
+     *
+     * <code> // violation
+     * CtMethod method = ... // violation
+     * CtRole role = CtRole.TYPE_MEMBER.getMatchingSubRoleFor(method); // violation
+     * </code> // violation
+     */
+    @Internal
+    public Object testMethod2() {
+        return new Object();
+    }
+}

--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/Example4.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/Example4.java
@@ -1,0 +1,50 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="JavadocTagContinuationIndentation"/>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadoctagcontinuationindentation;
+
+// xdoc section -- start
+
+class Example4 {
+
+   /**
+   * Writes the object using a
+   * <a href="{@docRoot}/serialized-form.html#java.time.Ser">dedicated form</a>.
+   * @serialData
+   * <pre>
+   * out.writeByte(1);
+   * out.writeInt(nanos);
+   * </pre>
+   */
+  public void testMethod1(String input) {}
+
+  /**
+   * Writes the object using a
+   * <a href="{@docRoot}/serialized-form.html#java.time.Ser">dedicated form</a>.
+   * @serialData
+   * <code> // violation
+   * out.writeByte(1); // violation
+   * out.writeInt(nanos); // violation
+   * </code> // violation
+   */
+  public void testMethod2(String input) {}
+
+  /**
+   * Test class.
+   *
+   * @apiNote
+   *          This is the predefined indentation applied by Eclipse formatter.
+   *          <pre>
+   * this content, and not any error:
+   *   "JavadocTagContinuation do not validate lines contained in Pre tag,
+   *   No violation is expected here."</pre>
+   */
+  public void testMethod3(String input) {}
+
+}
+
+// xdoc section -- end

